wgs_somatic-run-wrapper.py

```diff
@@ -150,18 +163,19 @@ def wrapper():
                             logger.info(f'Outputdir exists for {tumorsample}. Renaming old outputdir {outputdir} to {outputdir}_old')
                             os.rename(outputdir, f'{outputdir}_old')
 
-                        logger.info(f'Starting wgs_somatic with arguments: \n \
-runnormal: {runnormal} \n \
-runtumor: {runtumor} \n \
-tumorsample: {tumorsample} \n \
-normalsample: {normalsample} \n \
-normalfastqs: {normalfastqs} \n \
-tumorfastqs: {tumorfastqs} \n \
-outputdir: {outputdir} \n \
-igvuser: {igvuser} \n \
-hg38ref: {hg38ref}')
+                        pipeline_args = [['python', 'launch_snakemake.py', '--runnormal', f'{runnormal}', '--outputdir', f'{outputdir}', '--normalsample', f'{normalsample}', '--normalfastqs', f'{normalfastqs}', '--runtumor', f'{runtumor}', '--tumorsample', f'{tumorsample}', '--tumorfastqs', f'{tumorfastqs}', '--igvuser', f'{igvuser}', '--hg38ref', f'{hg38ref}']]
+
+                        # Using threading to start the pipeline for several samples at the same time
+                        threads.append(threading.Thread(target=call_script, args=pipeline_args))
+                        logger.info(f'Starting wgs_somatic with arguments {pipeline_args}')
+    for t in threads:
+        t.start()
+        logger.info(f'Thread {t} has started')
+    for u in threads:
+        u.join()
+        logger.info(f'Thread {u} is finished')
                         # Pass the correct arguments to launch_snakemake.py to start the pipeline
-                        subprocess.Popen(['python', 'launch_snakemake.py', '--runnormal', f'{runnormal}', '--outputdir', f'{outputdir}', '--normalsample', f'{normalsample}', '--normalfastqs', f'{normalfastqs}', '--runtumor', f'{runtumor}', '--tumorsample', f'{tumorsample}', '--tumorfastqs', f'{tumorfastqs}', '--igvuser', f'{igvuser}', '--hg38ref', f'{hg38ref}'])
```

Comment:
Maybe in the future we can change this to start the main function of launch_snakemake.py instead of launching a shell to run it

Reply:
solved in commit @ 51aa513